src/compiler/checker.ts

```diff
@@ -5666,7 +5655,24 @@ namespace ts {
                     }
                 }
 
-                return result & isRelatedTo(sourceReturnType, targetReturnType, reportErrors);
+                result = result & isRelatedTo(sourceReturnType, targetReturnType, reportErrors);
+                if (!(sourceReturnType.flags & TypeFlags.PredicateType && targetReturnType.flags & TypeFlags.PredicateType)) {
+                    return result;
+                }
+                const sourcePredicate = sourceReturnType as PredicateType;
+                const targetPredicate = targetReturnType as PredicateType;
+                if (sourcePredicate.predicate.kind === TypePredicateKind.Identifier) {
+                    const sourceIdentifierPredicate = sourcePredicate.predicate as IdentifierTypePredicate;
+                    const targetIdentifierPredicate = targetPredicate.predicate as IdentifierTypePredicate;
+                    if (sourceIdentifierPredicate.parameterIndex !== targetIdentifierPredicate.parameterIndex) {

```

Comment:
Do the parameterIndex and the predicate kind need to be on the type predicate anymore? Can they just be on the signature?